pubsub.go

```diff
@@ -621,6 +614,43 @@ func (p *PubSub) processLoop(ctx context.Context) {
 	}
 }
 
+func (p *PubSub) handlePendingPeers() {
+	select {
+	case <-p.newPeersSema:
+		defer func() {
+			p.newPeersSema <- struct{}{}
+		}()
+
+	default:
+		// contention, return and wait for the next notification without blocking the event loop
+		return
+	}
+
+	if len(p.newPeersPend) == 0 {
+		return
+	}
+
+	newPeers := p.newPeersPend
+	p.newPeersPend = make(map[peer.ID]struct{})
+
```

Comment:
Can we release the "lock" right here? There's no reason to hold it.

Reply:
sure